kb/how-to/rewrite-branch-for-coding-style.md

```diff
@@ -0,0 +1,163 @@
+# Switching to the new code style
+
+In January 2023, Mbed TLS switched to a new coding style, which is now enforced by continuous integration tests. This document explains how to adapt ongoing work. Please read this document if you started work on a branch of Mbed TLS that has the old coding style, and you want to contribute this work to Mbed TLS or you are already doing so.
+
+This concerns both maintained branches of Mbed TLS: `development` and `mbedtls-2.28`.
+
+## Coding style overview
+
+The following table summarizes the old and new coding style (before/after January 2023).
+
+| Aspect | Old | New |
+| ------ | --- | --- |
+| Indentation | 4 spaces | 4 spaces |
+| Indentation of `case` | one level deeper than `switch` | one level deeper than `switch` |
+| Maximum line length | 80 preferred | 80 preferred |
+| **Spacing inside parentheses** | always except `define` and casts: `if( f( x ) )` | **never**: `if (f(x))` |
+| Spacing inside square brackets | no: `a[i]` | no: `a[i]` |
+| **Spacing before opening parenthesis** | no: `if( f( x ) )` | **after keyword or binary operator**: `if (f(x))` |
+| Spacing before opening square bracket | no: `a[i]` | no: `a[i]` |
+| Spacing after unary prefix operator | optional: `!c` or `! c` | **no**: `!c` |
+| Spacing around binary operator | yes: `( x + y ) * z` | yes: `(x + y) * z` |
+| Parentheses around return value | mandatory: `return( x );` | **generally not**: `return x;` |
+| **Single-line if/loop body** | separate line, braces optional | separate line, **braces mandatory** |
+| Opening brace | on its own line | **on the same line** except for functions |
+
+For a more detailed description of the coding style, please refer to the [coding standards](../development/mbedtls-coding-standards.md).
+
+## Migrating existing work: the theory
```

Comment:
Minor: this makes me want to look for an "in practice" section, but it's called "how to" (which I agree nicely echoes the preceding "when to").